LEDGAME.ino

```diff
@@ -75,140 +79,216 @@ int randomGenTime()
   return randomtime;
 }
 
- //turn on random led
-int randomLedOn(int num)
+ //turn on led
+int ledOn(int num)
 {
-   digitalWrite(ledInit[num], HIGH);   
+  digitalWrite(ledInit[num], HIGH);   
 }
 
-
 //turn off all leds
 void ledOff(int num)
+{
+  digitalWrite(ledInit[num], LOW);
+}  
+
+//butt buffer 
+int checkButtPress()
+{  
+  for (int i = 1; i < BUTTONS_QUANTITY; i++)
   {
-    digitalWrite(ledInit[num], LOW);
+    if (digitalRead(buttInit[i]) == HIGH)
+    {
+      buttState[i] = 1;
+      buttPressStatus = true;
+    }
   }  
+}
 
-//буфер с нажатыми кнопками
-int WhichButtPressed()
-{
-  
-  for (int c = 1; c < 4; c++)
+int checkResult(int num)
+{ 
+ if (buttState[num] == 1 && buttPressStatus)
+  {
+    score += 1;
+    lcd.clear();
+    lcd.print("    CORRECT!");
+  }
+ else if (buttState[num] == 0 && buttPressStatus)
+  {
+    lcd.clear();
+    lcd.print("   INCORRECT(");
+  }
+ else if (!ledTimer.isTimerRunning() && !buttPressStatus)
   {
-   if (digitalRead(buttInit[c]) == HIGH)
+    lcd.clear();
+    lcd.print("NO BUTTON PRESS(");
+  }
+}
+
+//clearing butt bufer
+void clearButt()
+{
+  buttState[0] = 1; 
+  buttPressStatus = false;   
+  for (int i = 1; i < BUTTONS_QUANTITY; i++)
    {
-     butt[c] = c;
-     i = 1;
+     buttState[i] = 0;
    }
-  } 
-  
 }
 
-
-int Logic(int num)
+ //waiting for start button
+void startButtonWait()
 {
- if (butt[num] == num && i == 1)
- {
-  score += 1;
-  Serial.print("                                                                  CORRECT! \n");
-  //correct
- }
-  else if (butt[num] != num && i == 1)
+  while (buttState[0] == 0) 
   {
-    Serial.print("                                                                 INCORRECT( \n");
-    //incorrect
+    if(digitalRead(buttInit[0]) == HIGH)
+    { 
+      buttState[0] = 1; 
+    }
   }
-   else if (ledTimer.isTimerRunning() == false && i == 0)
-   {
-    Serial.print("                                                          NO BUTTON PRESS DETECTED \n");
-   }
 }
 
-//clearing butt bufer
-    void clearButt()
+void userSetGameTime()
+{
+  lcd.clear();
+  lcd.print(" set game time!");
+  lcd.setCursor(0, 1);
+  lcd.print("1=10, 2=20, 3=30");
+  while( !buttPressStatus )
+  {
+    checkButtPress();
+    for (unsigned long i = 1, t = 10000; i < BUTTONS_QUANTITY; i++)
     {
-        butt[0] = 1; 
-      for (int i = 1; i < 4; i++)
-      {
-        butt[i] = 0;
+      if (buttState[i] == 1)
+      {  
+        lcd.clear();
+        lcd.print("  setted time:");
+        lcd.setCursor(7, 1);
+        lcd.print(i);
+        lcd.setCursor(8, 1);
+        lcd.print("0!");
+        lcdTimer.setTimer(2000);
+        lcdTimer.delayTimer();
+        countdown();
+        timer.setTimer(t*i); //set how long game will last 
       }
-    }
+    } 
+  }
+}
 
- //waiting for start button
-void startButtonWait()

```

Comment:
function startButtonWait is OK, but it puts us into the eternal cycle - if no button pressed - we will wait forever)). Maybe it's better to use some timeout and when it expires - we go to the some LOGO-screen or whatever or show the previous scores. The timeout value can be several minutes or hours.